syntax = "proto3";

package inference;

service InferenceService {
  // Runs inference and returns predictions
  rpc RunInference (InferenceRequest) returns (InferenceResponse);
  
  // Health check for gRPC API
  rpc HealthCheck (HealthRequest) returns (HealthResponse);
}

message InferenceRequest {
  repeated float input = 1;  // List of input values
}

message InferenceResponse {
  repeated float result = 1; // Model predictions
  string status = 2; // "success" or "error"
  string message = 3; // Error message (if applicable)
}

message HealthRequest {}

message HealthResponse {
  string status = 1; // "ok"
  string message = 2; // Health status message
}
